includes/Stories_Renderer/Generic_Renderer.php

```diff
@@ -0,0 +1,389 @@
+<?php
+/**
+ * Generic_Renderer class.
+ *
+ * @package   Google\Web_Stories
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\Stories_Renderer;
+
+use Google\Web_Stories\Embed_Base;
+use Google\Web_Stories\Story_Post_Type;
+
+/**
+ * Generic_Renderer class.
+ */
+class Generic_Renderer extends Renderer {
+	/**
+	 * Perform initial setup for object.
+	 *
+	 * @return void
+	 */
+	public function init() {
+		parent::init();
+		add_action( 'web_stories_renderer_container_start', [ $this, 'amp_carousel' ] );
+		add_action( 'web_stories_renderer_container_end', [ $this, 'amp_carousel' ] );
+		add_action( 'web_stories_renderer_content', [ $this, 'render_story' ] );
+		add_action( 'web_stories_renderer_wrapper_end', [ $this, 'maybe_render_archive_link' ] );
+		add_action( 'web_stories_renderer_story_content_overlay', [ $this, 'get_content_overlay' ] );
+		add_action( 'web_stories_renderer_single_story_content', [ $this, 'render_story_with_poster' ] );
+		add_action( 'web_stories_renderer_single_story_content', [ $this, 'render_story_with_story_player' ] );
+		add_action( 'web_stories_renderer_container_classes', [ $this, 'container_classes' ] );
+		add_action( 'web_stories_renderer_container_style', [ $this, 'container_styles' ] );
+		add_action( 'web_stories_renderer_single_story_classes', [ $this, 'single_story_classes' ] );
```

Comment:
Re-iterating on my initial statement:

Templating by doing `add_action` and `current_action` checks everywhere is very hard to follow and maintain. For example, when reading the code it's hard to find out what the final carousel template looks like.